### ISSUE TITLE
Howto verify does enum exist by name

### ISSUE THREAD

**Issue by greyblake** [albertosaurus/power_enum#16]
Is there a method to verify that enum exist by name?

A few months ago `[]` used to return nil when String was passed:

```
Book["Hobbit"]  # => nil, there is no Hobbit enum
```

not it raises exception:

```
Book["Hobbit"] # => ActiveRecord::RecordNotFound: Couldn't find a Book identified by ("Hobbit")
```

**Comment by albertosaurus** [comment]
This behavior is specified by an option you pass in when you declare the enum called `on_lookup_failure`.  From the README/RDoc:

```
The :on_lookup_failure option specifies the name of a class method to invoke when the [] method is unable to locate a BookingStatus record for arg. The default is the built-in :enforce_none which returns nil. There are also built-ins for :enforce_strict (raise and exception regardless of the type for arg), :enforce_strict_literals (raises an exception if the arg is a Fixnum or Symbol), :enforce_strict_ids (raises and exception if the arg is a Fixnum) and :enforce_strict_symbols (raises an exception if the arg is a Symbol).

The purpose of the :on_lookup_failure option is that a) under some circumstances a lookup failure is a Bad Thing and action should be taken, therefore b) a fallback action should be easily configurable. As of version 0.8.4, you can also set :on_lookup_failure to a lambda that takes in a single argument (The arg that was passed to []).
```

Having said that, I think you're right that there needs to be a mechanism to readily determine if an item is a member of the enumeration.  Let me think about this.

**Comment by albertosaurus** [comment]
This has been implemented in 0.10.0.  You can do:

``` ruby
# Assume you have a collection of three books
# * 1 => 'Lord of the Rings'
# * 2 => '50 Shades of Grey'
# * 3 => 'Atlas Shrugged'

Book.contains? 'Hobbit' # returns false
Book.contains? 'Atlas Shrugged' # returns true
Book.contains? 2 # returns true
Book.contains? 999 # returns false
Book.contains? :'50 Shades of Grey' # returns true
```

**Comment by greyblake** [comment]
Thanks @albertosaurus!
